src/backend/dt.c

```diff
@@ -300,6 +300,98 @@ dt_t **dtdtoff(dt_t **pdtend, dt_t *dt, unsigned offset)
     return dtxoff(pdtend, s, offset);
 }
 
+/**************************************
+ * Repeat a list of dt_t's len times.
+ */
+dt_t **dtrepeat(dt_t **pdtend, dt_t *dt, size_t len)
+{
+    unsigned size = dt_size(dt);
+
+    if (dtallzeros(dt))
+        return dtnzeros(pdtend, size * len);
+
+    while (*pdtend)
+        pdtend = &((*pdtend)->DTnext);
+
+    if (len == 0)
+        return pdtend;
+
+    if (dtpointers(dt))
+    {
+        dt_t *dtp = NULL;
+        dt_t **pdt = &dtp;
+        for (size_t i = 0; i < len; ++i)
+        {
+            for (dt_t *dtn = dt; dtn; dtn = dtn->DTnext)
+            {
+                dt_t *dtx = dt_calloc(dtn->dt);
+                *dtx = *dtn;
+                dtx->DTnext = NULL;
+                switch (dtx->dt)
+                {
+                    case DT_abytes:
+                    case DT_nbytes:
+                        dtx->DTpbytes = (char *) MEM_PH_MALLOC(dtx->DTnbytes);
+                        memcpy(dtx->DTpbytes, dtn->DTpbytes, dtx->DTnbytes);
+                        break;
+                }
+
+                *pdt = dtx;
+                pdt = &dtx->DTnext;
+            }
+        }
+        *pdtend = dtp;
+        return pdt;
+    }
+    else
```

Comment:
I think it would be clearer without the `else` block, as the block and the rest of the function will always be run together.